xds/src/main/java/io/grpc/xds/XdsLoadBalancer2.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.grpc.ChannelLogger.ChannelLogLevel;
+import io.grpc.ConnectivityState;
+import io.grpc.ConnectivityStateInfo;
+import io.grpc.LoadBalancer;
+import io.grpc.Status;
+import io.grpc.SynchronizationContext.ScheduledHandle;
+import io.grpc.util.ForwardingLoadBalancerHelper;
+import io.grpc.xds.XdsComms.AdsStreamCallback;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+
+/**
+ * A {@link LoadBalancer} that uses the XDS protocol.
+ *
+ * <p>This class manages fallback handling. The logic for child policy handling and fallback policy
+ * handling is provided by LookasideLb and FallbackLb.
+ */
+// TODO(zdapeng): migrate name to XdsLoadBlancer
+final class XdsLoadBalancer2 extends LoadBalancer {
+
+  private static final long FALLBACK_TIMEOUT_MS = TimeUnit.SECONDS.toMillis(10); // same as grpclb
+
+  private final Helper helper;
+  private final LoadBalancer lookasideLb;
+  private final LoadBalancer.Factory fallbackLbFactory;
+  private final AdsStreamCallback adsCallback = new AdsStreamCallback() {
```

Comment:
This is `FallbackHandler` with `onBalancerWorking()`, `onBalancerError()` and `onBalancerAllDrop()` as Fallback balancer is the implementer and the handler is injected into lookaside balancer. This also makes the naming more semantically meaningful.

Reply:
`AdsStreamCallback` vs `FallbackHandler` was disscussed previously https://github.com/grpc/grpc-java/pull/5356/commits/1483cd356725f75d06b13609e0b3c50beceb943e#r257041208 and offline. Beside this, it will not only be used for fallback, `lrsClient` will also wrap/add listener to it.